deltacat/compute/compactor_v2/steps/hash_bucket.py

```diff
@@ -179,28 +186,37 @@ def _timed_hash_bucket(input: HashBucketInput):
 
 @ray.remote
 def hash_bucket(input: HashBucketInput) -> HashBucketResult:
+    with NodeUtilizationOverTimeRange() as node_util:
+        logger.info(f"Starting hash bucket task...")
 
-    logger.info(f"Starting hash bucket task...")
-    hash_bucket_result, duration = timed_invocation(
-        func=_timed_hash_bucket, input=input
-    )
+        # Log node peak memory utilization every 10 seconds
+        def log_peak_memory():
+            logger.debug(
+                f"Node peak memory utilization so far: {node_util.max_memory} ({node_util.max_memory/BYTES_PER_GIBIBYTE} GB)"
+            )
+
+        node_util.schedule_callback(log_peak_memory, 10)
```

Comment:
How would you coordinate that? Then each process would have to know whether other processes already started, right?

Also, I think the outputted value is process-specific. In testing, I can see many different values outputted by different task IDs, so it's not a shared value across all tasks.

Reply:
Is it the memory used by process or the entire node by all processes?

Reply:
https://docs.python.org/3/library/resource.html#resource.getrusage

Unfortunately, it looks like it's process specific

Reply:
That doesn't give us what we are looking for. Does it?

Reply:
I'd view it as more of a proxy for memory usage. While we won't know the exact utilization of the entire node's memory, this change lets us see if a process if using significantly more memory than expected. I'm not sure if there are additional complications wrt virtual memory vs actual physical memory, but I don't think it's useless either.

Reply:
Sure, let's rename it `TaskLevelNodeUtilizationOverTime`.

Reply:
I've already renamed it to `ProcessUtilizationOverTimeRange`, I think it's a fine name because it's very explicit about what is being measured. If you have strong opinions I could change the name but I think this is ready to merge.